include/swift/Driver/Action.h

```diff
@@ -323,6 +324,22 @@ class LinkJobAction : public JobAction {
   }
 };
 
+class ArchiveJobAction : public JobAction {
+  virtual void anchor();
+
+public:
+  ArchiveJobAction(ArrayRef<const Action *> Inputs, LinkKind K)
+      : JobAction(Action::Kind::ArchiveJob, Inputs, file_types::TY_Image) {
+    assert(K == LinkKind::StaticLibrary);
+  }
+
+  LinkKind getKind() const { return LinkKind::StaticLibrary; }
```

Comment:
You probably don't need this. I appreciate it being passed in to the constructor, though.